bears/python/requirements/PySafetyBear.py

```diff
@@ -26,6 +29,18 @@ def cve_key_checker(vulnerability):
 Package = namedtuple('Package', ('key', 'version'))
 
 
+def _get_vulnerabilities(pkg, spec, db):
+    for entry in db[pkg]:
+        entry['cve'] = 'None' if entry['cve'] is None else entry['cve']
+        entry['id'] = entry['cve']
+        for entry_spec in entry['specs']:
+            if entry_spec == spec:
+                yield entry
+
+
+safety.get_vulnerabilities = _get_vulnerabilities
```

Comment:
add the `cve` in a hack of `get_database_file` instead of a hack of `get_vulnerabilities`.

You are already doing a hack on their db loading mechanism, in `setup_dependencies`, so it is much safer and logical to modify their file loading mechanism than to duplicate&modify their function which has comparison logic.

The hack should be in `setup_dependencies` .

And to help with future breakages, you should first do `safety.get_database_file` (EOL) , as that will make Python check that they have a function called `safety.get_database_file`.  If it doesnt, the exception will occur there.  Then you can replace it, knowing the safety code will call it.

Otherwise they may change the name of that function and the bear error will be very confusing.

Reply:
ah, it is called `fetch_database_file`

Reply:
and alternative is to keep hacking `get_vulnerabilities` , but also use it!

```
safety_get_vulnerabilities = safety.get_vulnerabilities

def _get_vulnerabilities(...)
    for entry in safety_get_vulnerabilities(...):
        hack_entry
        yield entry
```